src/pages/SettingsPage/UsersSettings/newServiceUser.jsx

```diff
@@ -0,0 +1,176 @@
+import { useRef, useState } from 'react'
+import { toast } from 'react-toastify'
+import styled from 'styled-components'
+import { SelectButton } from 'primereact/selectbutton'
+
+import { Button, SaveButton, Section, Dialog, FormRow } from '@ynput/ayon-react-components'
+import ApiKeyManager from '@/components/ApiKeyManager'
+import useUserMutations from '@/containers/Feed/hooks/useUserMutations'
+import { useAddUserMutation } from '@queries/user/updateUser'
+import copyToClipboard from '@/helpers/copyToClipboard'
+import callbackOnKeyDown from '@/helpers/callbackOnKeyDown'
+
+import UserAttribForm from './UserAttribForm'
+import { uniqueId } from 'lodash'
+
+const FormRowStyled = styled(FormRow)`
+  .label {
+    min-width: 160px;
+  }
+`
+
+const NewServiceUser = ({ onHide, open, onSuccess }) => {
+  const {
+    password, setPassword,
+    formData, setFormData,
+    apiKey, setApiKey,
+    addedUsers, setAddedUsers,
+  } = useUserMutations({})
+
+
+  const [addUser, { isLoading: isCreatingUser }] = useAddUserMutation()
+  const usernameRef = useRef()
+  const [keyName, setKeyName] = useState('');
+
+  const validateFormData = (formData) => {
+    if (!formData.Username) {
+      return 'Login name must be provided';
+    }
+    if (apiKey == '') {
+      return 'Api key needs to be generated'
+    }
+
+    return null;
+  }
+
+  const resetFormData = ({ addedUsers }) => {
+    setFormData({
+      Username: '',
+      userActive: true
+    })
+    setAddedUsers(addedUsers)
+  }
+
+  const preparePayload = (formData, apiKey) => {
+    const payload = {
+      data: { isService: true },
+      active: formData.userActive,
+      name: formData.Username,
+      apiKey: apiKey
+    }
+
+    return payload;
+  }
+
+  const handleSubmit = async (close) => {
+    const validationResult = validateFormData(formData);
+    if (validationResult !== null) {
+      toast.error(validationResult)
+      return
+    }
+
+    try {
+      await addUser({ name: formData.Username, user: preparePayload(formData, apiKey) }).unwrap()
+      toast.success('Service User created')
+
+      resetFormData({
+        addedUsers: [...addedUsers, formData.Username]
+      });
+
+      setKeyName(uniqueId())
+
+      onSuccess && onSuccess(formData.Username)
+
+      if (close) {
+        onHide([formData.Username])
+      } else {
+        usernameRef.current?.focus()
+      }
+    } catch (error) {
+      console.error(error)
+      toast.error(`Unable to create user: ${error.detail}`)
+    }
+  }
+
+  const handleClose = () => {
+    resetFormData({
+      addedUsers: [],
+    })
+    onHide(addedUsers)
+  }
+
+  const handleApiKeyGeneration = (key) => {
+    copyToClipboard(key);
+    setApiKey(key);
+  }
+
+  if (!open) return null
+
+  return (
+    <Dialog
+      onKeyDown={(e) => callbackOnKeyDown(e, {
+        validationPassed: formData.Username && password,
+        callback: handleSubmit,
+      })}
+      isOpen
```

Comment:
I was trying to work out why cmd+enter wasn't working and realised that it's checking for a password. There is no password.
```suggestion
          validationPassed: formData.Username,
```